mojap_metadata/metadata/metadata.py

```diff
@@ -177,11 +177,26 @@ def from_yaml(cls, filename, **kwargs) -> object:
             obj = yaml.safe_load(f, **kwargs)
             return cls.from_dict(obj)
 
+    @classmethod
+    def from_infer(cls, inp, **kwargs) -> object:
```

Comment:
```suggestion
    def from_infer(cls, inp, **kwargs) -> Metadata:
```

Reply:
Can't do that, it would be self referential. Also can't do `-> cls`

Reply:
Ah interesting